src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,34 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws ImpossibleRegisterUserException {
+        if (user == null) {
+            throw new ImpossibleRegisterUserException("The user can`t be empty");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new ImpossibleRegisterUserException("This login is incorrect");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new ImpossibleRegisterUserException("This password is incorrect");
+        } else if (user.getAge() == null || user.getAge() < 18) {
+            throw new ImpossibleRegisterUserException("Age of the user isn`t enough to show content");
+        } else {
+            List<User> copy = new ArrayList<>(Storage.people);
+            for (int i = 0; i < copy.size(); i++) {
+                if (copy.get(i).getLogin().equals(user.getLogin())) {
+                    throw new ImpossibleRegisterUserException("The user who have this login already exists");
+                }
+            }
```

Comment:
The logic for checking if a user with the same login already exists involves creating a copy of the `Storage.people` list and iterating over it. Consider optimizing this by directly iterating over `Storage.people` if the task requirements allow it, as creating a copy may be unnecessary.